### PR TITLE
Add recurring tasks

### PR DESCRIPTION
**Pull request by JasonScourge**

### CODE REVIEW

**Comment by JasonScourge** [comment]
There is a new in-built Recurring Category that needs can be removed if need be

File: src/main/java/seedu/doerList/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -12,17 +13,17 @@
...
 
 
     public static final String COMMAND_WORD = "add";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a task to the Do-erlist. "
-            + "Parameters: /t TASK [/d DESCRIPTION] [/s START] [/e END] [/c [CATEGORY] ... \n"
+            + "Parameters: /t TASK [/d DESCRIPTION] [/s START] [/e END] [/c CATEGORY] [/r] ... \n"
             + "Example: " + COMMAND_WORD
-            + " /t Take lecture /s 2016-10-04 10:00 /e 2016-10-04 12:00 /c CS2102";
+            + " /t Take lecture /s 2016-11-25 10:00 /e 2016-11-26 12:00 /c CS2102 /r recurring";
```
Comment (xpdavid) [inline src/main/java/seedu/doerList/logic/commands/AddCommand.java]: recurring need data time interval

File: src/main/java/seedu/doerList/logic/commands/MarkCommand.java [hunk 2]
```diff
@@ -1,40 +1,112 @@
...
-        
+
         try {
             ReadOnlyTask target = TaskListPanel.getDisplayedIndexWhenCategorizedByBuildInCategory(targetIndex, lastShownList);
-            model.markTask(target);
-            return new CommandResult(String.format(MESSAGE_MARK_TASK_SUCCESS, target));  
+            if (target.hasRecurring()){ 
+                Task newTask = generateNewTask(target);
+                updateRecurringTask(newTask);
+                model.changeTask(target, newTask);
```
Comment (xpdavid) [inline src/main/java/seedu/doerList/logic/commands/MarkCommand.java]: I think model has `replaceTask`

File: src/main/java/seedu/doerList/logic/commands/MarkCommand.java [hunk 3]
```diff
@@ -1,40 +1,112 @@
...
+
     }
+
+    /**
+     * Generate new task based on on read only task information
+     *
+     * @param original Task (ReadOnlyTask before update)
+     * @return the Task itself
+     */
+    private Task generateNewTask(ReadOnlyTask original) {
```
Comment (xpdavid) [inline src/main/java/seedu/doerList/logic/commands/MarkCommand.java]: The constructor of `Task` itself has such copy function.

File: src/main/java/seedu/doerList/model/DoerList.java [hunk 4]
```diff
@@ -169,6 +173,15 @@ public void markTask(ReadOnlyTask task) throws UniqueTaskList.TaskNotFoundExcept
             throw new UniqueTaskList.TaskNotFoundException();
         }
     }
+    
+    //@@author A0139401N
+    public void recurTask(ReadOnlyTask task) throws UniqueTaskList.TaskNotFoundException {
```
Comment (xpdavid) [inline src/main/java/seedu/doerList/model/DoerList.java]: No need this anymore right?

File: src/main/java/seedu/doerList/ui/TaskListPanel.java [hunk 5]
```diff
@@ -35,7 +35,7 @@
     public static final BuildInCategory[] categorizedBy = {
             BuildInCategoryList.DUE,
             BuildInCategoryList.TODAY, BuildInCategoryList.NEXT, BuildInCategoryList.INBOX,
-            BuildInCategoryList.COMPLETE
+            BuildInCategoryList.COMPLETE, BuildInCategoryList.RECURRING
```
Comment (xpdavid) [inline src/main/java/seedu/doerList/ui/TaskListPanel.java]: I think we just add a small badge to the task card instead of adding it to a new section.

Comment (JasonScourge) [inline src/main/java/seedu/doerList/ui/TaskListPanel.java]: Yeah, thanks for pointing that out. 
Will remedy that.

File: src/main/java/seedu/doerList/logic/commands/MarkCommand.java (hunk 2 above)
Comment (JasonScourge) [inline src/main/java/seedu/doerList/logic/commands/MarkCommand.java]: One of the previous commits I use replaceTask, keeps giving me duplicateTask error each time I try to create a recurring task.

File: src/main/java/seedu/doerList/logic/commands/MarkCommand.java (hunk 3 above)
Comment (JasonScourge) [inline src/main/java/seedu/doerList/logic/commands/MarkCommand.java]: Ah thanks for pointing that out.

File: src/main/java/seedu/doerList/model/DoerList.java (hunk 4 above)
Comment (JasonScourge) [inline src/main/java/seedu/doerList/model/DoerList.java]: Will remove this.

File: src/main/java/seedu/doerList/model/task/TodoTime.java [hunk 6]
```diff
@@ -39,7 +39,8 @@ public LocalDateTime getTime() {
        return value;
    }
    
-   //@@author A0139401N
+   // I changed this method, where is this method used though? 
+   // Don't seem to see it anywhere
```
Comment (JasonScourge) [inline src/main/java/seedu/doerList/model/task/TodoTime.java]: If this method is not being used, shall we remove it?

File: src/main/java/seedu/doerList/logic/commands/AddCommand.java (hunk 1 above)
Comment (JasonScourge) [inline src/main/java/seedu/doerList/logic/commands/AddCommand.java]: Remedied

File: src/main/java/seedu/doerList/logic/commands/MarkCommand.java (hunk 3 above)
Comment (JasonScourge) [inline src/main/java/seedu/doerList/logic/commands/MarkCommand.java]: Remedied

File: src/main/java/seedu/doerList/ui/TaskListPanel.java (hunk 5 above)
Comment (JasonScourge) [inline src/main/java/seedu/doerList/ui/TaskListPanel.java]: Remedied

**Pull request merged**